jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/traceformat/TraceFormat.java

```diff
@@ -355,21 +356,24 @@ public static void main(String[] args) throws Exception {
 				thread = current;
 				output.println(formatted.toString());
 			}
-			
+
 			/* print percentage */
 			if (context.getTotalRecords() != recordsProcessed) {
 				recordsProcessed = context.getTotalRecords();
-				float processedMbytes = ((float) (context.getTotalRecords() * context.getRecordSize())) / (1024 * 1024);
+				long bytesProcessed = recordsProcessed * context.getRecordSize();
+				float processedMbytes = bytesProcessed / (1024.0f * 1024.0f);
 				if (processedMbytes % 10 == 0) {
-					int percent = (int) ((processedMbytes * 100) / (totalBytes/(1024*1024)));
+					int percent = (int) ((100 * bytesProcessed) / totalBytes);
 					if (verbose.booleanValue()) {
-						end = System.currentTimeMillis();
-						float MbpsBlock = (float) 10.0 / (float) ((end - startBlock) / 1000.0);
-						float Mbps = (float) (processedMbytes) / (float) (((float) (end - start) / 1000.0));
-						startBlock = System.currentTimeMillis();
-						context.message(context, "Processed " + processedMbytes + "Mb ("+percent+"%), burst speed: "+MbpsBlock+"Mb/s, average: "+Mbps+"Mb/s");
+						end = System.nanoTime();
+						final long SECOND_SCALE = 1000 * 1000 * 1000;
+						float MbpsBlock = 10.0f * SECOND_SCALE / (end - startBlock);
+						float Mbps = processedMbytes / (end - start) * SECOND_SCALE;
```

Comment:
Please reorder the operations for better precision:
```
    float Mbps = processedMbytes * SECOND_SCALE / (end - start);
```

Reply:
Updated.